pico/tests/interception/src/test/java/io/helidon/pico/tests/interception/InterceptionTest.java

```diff
@@ -0,0 +1,151 @@
+                () -> assertThat("Interceptors should not be called for method not annotated",
+                                 ModifyingInterceptor.lastCall(),
+                                 nullValue()),
+                () -> assertThat("Interceptors should not be called for method not annotated",
+                                 RepeatingInterceptor.lastCall(),
+                                 nullValue())
+        );
+
+        assertThat(response, is("hello"));
+    }
+
+    @Test
+    void testReturn() {
+        String response = service.intercepted("hello", false, false, true);
+
+        Invocation last = ReturningInterceptor.lastCall();
+
+        assertAll(
+                () -> assertThat("Returning last call", last, notNullValue()),
+                () -> assertThat("Returning last call", last.methodName(), is("intercepted")),
+                () -> assertThat("Returning last call", last.args(), is(new Object[] {"hello", false, false, true})),
+                () -> assertThat("Interceptors should not be called when first interceptor returns",
+                                 ModifyingInterceptor.lastCall(),
+                                 nullValue()),
+                () -> assertThat("Interceptors should not be called when first interceptor returns",
+                                 RepeatingInterceptor.lastCall(),
+                                 nullValue())
+        );
+
+        assertThat(response, is("fixed_answer"));
+    }
+
+    @Test
+    void testModify() {
+        String response = service.intercepted("hello", true, false, false);
+        assertThat(response, is("mod_hello"));
+
+        Invocation returning = ReturningInterceptor.lastCall();
+        Invocation modifying = ModifyingInterceptor.lastCall();
+        Invocation repeating = RepeatingInterceptor.lastCall();
+
+        assertAll(
+                () -> assertThat("Returning last call", returning, notNullValue()),
+                () -> assertThat("Returning last call", returning.methodName(), is("intercepted")),
+                () -> assertThat("Returning last call", returning.args(), is(new Object[] {"hello", true, false, false})),
+                () -> assertThat("Modifying last call", modifying, notNullValue()),
+                () -> assertThat("Modifying last call", modifying.methodName(), is("intercepted")),
+                () -> assertThat("Modifying last call", modifying.args(), is(new Object[] {"hello", true, false, false})),
+                () -> assertThat("Repeating last call", repeating, notNullValue()),
+                () -> assertThat("Repeating last call", repeating.methodName(), is("intercepted")),
+                () -> assertThat("Repeating last call", repeating.args(), is(new Object[] {"mod_hello", true, false, false}))
+        );
+    }
+
+    @Disabled("Known problem - issue #6629")
+    @Test
+    void testRepeat() {
+        String response = service.intercepted("hello", false, true, false);
```

Comment:
instead of all of these boolean switches, can we just use a Builder arg - this will be more flexible for other testing scenarios...

Reply:
I would prefer to use an array of parameters, to make sure the parameters are correctly handled in the array

Reply:
Also adding builder is testing an additional component - this is as much "clean" test for interceptors as possible.